### ISSUE TITLE
[Paper Note] Training nGPT, Ilya Loshchilov+, arXiv'26, 2026.08

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#6265]
# URL
- https://arxiv.org/abs/2608.01284
# Authors
  - Ilya Loshchilov
  - Boris Ginsburg
# Abstract
  - The normalized Transformer (nGPT) realizes hyperspherical representation learning by constraining model parameter vectors and activation vectors to the unit hypersphere. In this paper, we describe a practical training recipe for nGPT and evaluate it on modern hybrid Mamba-2--Transformer Mixture-of-Experts (MoE) models. The recipe introduces Logit Gradient Preconditioning, Logarithmic Learning Rate Decay, GatedAdamW, angular update control, and optional exploration mechanisms. Compared with an unnormalized model of the same hybrid MoE architecture trained with AdamW, the 14B-total-parameter nGPT model reaches the same validation loss using approximately half as many training tokens. The recipe scales across the models considered, which contain up to 14B total parameters.
# Translation (by gpt-5-nano)
- 正規化されたトランスフォーマー（nGPT）は、モデルパラメータベクトルと活性化ベクトルを単位超球面に制約することによって、超球面表現学習を実現します。本論文では、nGPTの実用的なトレーニングレシピを説明し、それを現代のハイブリッドMamba-2-Transformer Mixture-of-Experts（MoE）モデルで評価します。レシピには、ロジット勾配の前処理、対数学習率減衰、ゲーテッドAdamW、角度更新の制御、および任意の探索メカニズムが導入されています。同じハイブリッドMoEアーキテクチャをAdamWで訓練した未正規化モデルと比較すると、総パラメータが14BのnGPTモデルは、ほぼ半分のトレーニングトークン数で同じ検証損失に到達します。レシピは、検討対象のモデルで最大14Bの総パラメータを含むものに対してスケールします。
# Summary (by gpt-4o-mini)
- nGPTは、モデルパラメータと活性化を超球面に制約し、超球面表現学習を実現。実用的なトレーニングレシピを提案し、ハイブリッドMoEモデルで評価。ロジット勾配の前処理や学習率減衰などを導入し、未正規化モデルと比較して、少ないトレーニングトークンで同等のパフォーマンスを達成。検討対象モデルは最大14Bのパラメータにスケール可能。

**Comment by AkihikoWatanabe** [comment]
元ポスト: https://x.com/rosinality/status/2084535924986167520?s=20

**Comment by AkihikoWatanabe** [comment]
nGPT: 
- #1465

**Comment by AkihikoWatanabe** [comment]
解説:https://x.com/hillbig/status/2086584281493844252?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q

**Comment by AkihikoWatanabe** [comment]
nGPTを提案した著者らによる研究で、オリジナルのnGPTではdenseなアーキテクチャで、1Bまでのサイズでしか実験がされていなかった。それをより大規模な14BBレベル、かつMoE、Mamba-2とTransformerのハイブリッドモデルで実験し、学習のために必要な提案を実施しているようである。3.3節にどのような変更が必要か、サマリが記述されている。

性能は、通常のGPTと比較して一貫して改善しており興味深い。
<img width="1018" height="388" alt="Image" src="https://github.com/user-attachments/assets/c75f66f4-bc6f-4ea4-8d49-d030398931c5" />

ハイパーパラメータが増えているように感じるが、どの程度ハイパーパラメータの設定が効いてくるのだろうか。学習率の調整だけでも厄介なので、モデルの性能は魅力的でロマンがあるが、実務上は採用しづらい気はする。
